src/main/kotlin/study/racinggame/presentation/RaceController.kt

```diff
@@ -0,0 +1,24 @@
+package study.racinggame.presentation
+
+import study.racinggame.domain.Race
+import study.racinggame.presentation.view.DrawView
+import study.racinggame.presentation.view.InputView
+import study.racinggame.service.RaceRule
+import study.racinggame.service.reader.InputReader
+import kotlin.random.Random
+
+class RaceController(
+    inputReader: InputReader,
+    random: Random = Random
+) {
+
+    private val drawView: DrawView = DrawView()
+    private val inputView = InputView().initInputView(inputReader)
+    private val raceRule = RaceRule(random = random)
+
+    fun startRace() {
+        val race = Race(inputView.carStorage(), raceRule)
+        repeat(inputView.raceTryCnt()) { drawView.drawCarState(race.runTrack()) }
+        drawView.drawRaceWinner(race.getRaceWinner())
+    }
```

Comment:
```suggestion
class RaceController(
    inputReader: InputReader
) {

    private val drawView: DrawView = DrawView()
    private val inputView = InputView().initInputView(inputReader)

    fun startRace() {
        val race = Race(inputView.carStorage())
        repeat(inputView.raceTryCnt()) { drawView.drawCarState(race.runTrack()) }
        drawView.drawRaceWinner(race.getRaceWinner())
    }
```

raceRule도 domain에 대한 부분이라고 생각합니다. 그렇기 때문에 raceRule이라는 클래스가 RaceController가 만들어지기 전에 주입이 안되도 괜찮아 보입니다. 추가로 Race 안에 RaceRule이 defaultValue로 되어있다면, Controller 코드가 더 좋아지지 않을까 생각하네요.

Reply:
컨트롤러에서 가짜 객체를 주입하여서 테스트 하기 위해서 raceRule를 주입 받고 있었습니다.

잘 생각해보니 컨트롤러 테스트에는 Random 값을 가짜 객체로 만들어서 강제로 원하는 결과를 호출시키는 것보다는

컨트롤러가 정상 작동하여 결과물을 표출하는 특정 메소드가 잘 불러오는지를 테스트하는 것이 더 잘 맞는다고 생각되네요.

가짜 Output 객체를 만들어서 수정해보겠습니다!